src/imitation/data/buffer.py

```diff
@@ -65,41 +65,70 @@ def __init__(
         self._idx = 0
 
     @classmethod
-    def from_data(cls, data: Dict[str, np.ndarray]) -> "Buffer":
-        """Constructs and return a Buffer containing only the provided data.
+    def from_data(
+        cls,
+        data: Dict[str, np.ndarray],
+        capacity: Optional[int] = None,
+        truncate_ok: bool = False,
+    ) -> "Buffer":
+        """Constructs and return a Buffer containing the provided data.
 
-        The returned Buffer is at full capacity and ready for sampling.
+        Shapes and dtypes are automatically inferred.
 
         Args:
             data: A dictionary mapping keys to data arrays. The arrays may differ
                 in their shape, but should agree in the first axis.
+            capacity: The Buffer capacity. If not provided, then this is automatically
+                set to the size of the data, so that the returned Buffer is at full
+                capacity.
+            truncate_ok: Whether to error if `capacity` < the number of samples in
+                `data`. If False, then only store the last `capacity` samples from
+                `data` when overcapacity.
+
+        Examples:
+            `Buffer` with same capacity as arrays in `data`::
```

Comment:
```suggestion
            In this example, suppose the arrays in `data` are length-1000.
            
            `Buffer` with same capacity as arrays in `data`::
```
I found it a bit ambiguous whether `10000` is larger capacity (assumes how big `data` is). Otherwise, example looks good :)